presenters/thread.rb

```diff
@@ -70,27 +71,29 @@ def to_hash with_responses=false, resp_skip=0, resp_limit=nil
   #     children)
   #   response_count
   #     The total number of responses
-  def get_paged_merged_responses(thread_id, responses, skip, limit)
+  def get_paged_merged_responses(thread_id, responses, skip, limit, recursive=false)
     response_ids = responses.only(:_id).sort({"sk" => 1}).to_a.map{|doc| doc["_id"]}
     paged_response_ids = limit.nil? ? response_ids.drop(skip) : response_ids.drop(skip).take(limit)
     content = Comment.where(comment_thread_id: thread_id).
       or({:parent_id => {"$in" => paged_response_ids}}, {:id => {"$in" => paged_response_ids}}).
       sort({"sk" => 1})
-    {"responses" => merge_response_content(content), "response_count" => response_ids.length}
+    {"responses" => merge_response_content(content, recursive), "response_count" => response_ids.length}
```

Comment:
@wajeeha-khalid @jcdyer I think this is an excellent idea. This section of code is the one area that has caused us real pain in production in the past, anything that we can do to bring less data back from mongo and to push any pagination down to the database level is a good idea.